File: sdk/csharp/src/IO.StockX.Examples/PortfolioAskExample.cs
using System;
using System.Diagnostics;
using IO.StockX.Api;
using IO.StockX.Client;
using IO.StockX.Model;

namespace Example
{
    public class PortfolioAskExample
    {
        static public void Main()
        {
            // Configure API key authorization: api_key
            Configuration.Default.AddApiKey("x-api-key", ExampleConstants.AWS_API_KEY);

            var stockx = new StockXApi();
                stockx.Configuration.BasePath = ExampleConstants.STOCKX_ENDPOINT;
            
            var login = new LoginRequest(ExampleConstants.STOCKX_USERNAME, ExampleConstants.STOCKX_PASSWORD);
            
            try
            {
                // Login and fetch the jwt header for authentication use in the request
                ApiResponse<LoginResponse> result = stockx.LoginWithHttpInfo(login);
                var jwt = result.Headers[ExampleConstants.JWT_HEADER];

                stockx.Configuration.DefaultHeader["jwt-authorization"] = jwt;

                var item = new PortfolioRequestPortfolioItem();
                    item.Amount = "25";
                    item.SkuUuid = "bae25b67-a721-4f57-ad5a-79973c7d0a5c";
                    item.MatchedWithDate = "2018-12-12T05:00:00+0000";
                    item.ExpiresAt = "2018-12-12T12:39:07+00:00";

                var request = new PortfolioRequest();
                    request.PortfolioItem = item;
                    request.Customer = result.Data.Customer;
                    request.Timezone = "America/Detroit";
                
                // Create a new portfolio ask
                var askResp = stockx.NewPortfolioAsk(request);

                // Print the result of the ask call
                Console.WriteLine(askResp);
            } 
            catch (Exception e)
            {
                Console.WriteLine(e.ToString());
                Console.WriteLine("Exception when calling StockXApi.NewPortfolioAsk: " + e.Message );
            }
        }
    }
}

File: sdk/java/io.stockx.sdk.examples/src/main/java/io/stockx/sdk/examples/endpoints/PortfolioAskExample.java
package io.stockx.sdk.examples.endpoints;

import static io.stockx.sdk.examples.base.LoginUtil.login;

import io.stockx.client.ApiClient;
import io.stockx.client.ApiException;
import io.stockx.client.api.StockXApi;
import io.stockx.client.model.CustomerObject;
import io.stockx.client.model.PortfolioRequest;
import io.stockx.client.model.PortfolioRequestPortfolioItem;
import io.stockx.client.model.PortfolioResponse;
import io.stockx.sdk.examples.base.Environment;
import io.stockx.sdk.examples.base.ExampleConstants;
import io.stockx.sdk.examples.base.StockXClient;

public class PortfolioAskExample {

    public static void main(String[] args) {

        ApiClient apiClient = StockXClient.create(ExampleConstants.AWS_API_KEY, Environment.STAGING, true);

        StockXApi stockx = new StockXApi(apiClient);

        try {
            // Login to the StockX framework - see LoginExample for how this works
            login(stockx, ExampleConstants.STOCKX_USERNAME, ExampleConstants.STOCKX_PASSWORD);

            PortfolioRequestPortfolioItem item = new PortfolioRequestPortfolioItem()
                            .amount("25")
                            .skuUuid("bae25b67-a721-4f57-ad5a-79973c7d0a5c")
                            .matchedWithDate("2018-12-12T05:00:00+0000")
                            .expiresAt("2018-12-12T12:39:07+00:00");

            CustomerObject customer = new CustomerObject();
                customer.setDefaultSize("7");

            PortfolioRequest request = new PortfolioRequest()
                            .portfolioItem(item)
                            .customer(customer)
                            .timezone("America/Detroit");

            PortfolioResponse askResp = stockx.newPortfolioAsk(request);

            System.out.println("Ask Created: " + askResp);
        } catch (ApiException e) {
            System.out.println(e.getMessage());
            // e.printStackTrace();
        }
    }

}
